keyboards/handwired/ms_sculpt_mobile/keymaps/milestogo/keymap.c

```diff
@@ -121,146 +164,308 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
-   			return MACRO_NONE;
-   		}
-   	}
-#endif
-
+    switch (keycode) {
+        case QWR:
+            if (record->event.pressed) {
+                layer_off(_CDH);
+            }
+            return false;
+            break;
 
-  // MACRODOWN only works in this function
-      switch(id) {
-        case 0:
-          if (record->event.pressed) {
-            register_code(KC_RSFT);
-          } else {
-            unregister_code(KC_RSFT);
-          }
-        break;
+        case CDH:
+            if (record->event.pressed) {
+                layer_on(_CDH);
+            }
+            return false;
+            break;
 
- /* Colemak mod-dh moves the D key to the qwerty V position
-   This hack makes apple-V_position do what I mean */
- 	case DHPASTE:
-		if(keyboard_report->mods & MOD_BIT(KC_LGUI) ) {
-			if (record->event.pressed) {
-	  	 		clear_keyboard_but_mods();
-		 		register_code(KC_V);
-	     	} else {
-		 		unregister_code(KC_V);
-	    	}
-		} else {
+        case SYM:
             if (record->event.pressed) {
-          		register_code(KC_D);
+                layer_on(_SYM);
             } else {
-           		unregister_code(KC_D);
-	   		}
-		}
-	break;
-
-	case VIBRK: // vi esc:
-		 if (record->event.pressed) {
-			return MACRO( T(ESC),D(LSFT),T(SCLN),U(LSFT), END );
-		 }
-	break;
-
-
-
-
-	default:
-    	return MACRO_NONE;
+              layer_off(_SYM);
+            }
+            return false;
+            break;
+
+        /* Colemak mod-dh moves the D key to the qwerty V position
+            This hack makes apple-V_position do what I mean */
+        case DHPASTE:
+            if(keyboard_report->mods & MOD_BIT(KC_LGUI) ) {
+                if (record->event.pressed) {
+                    clear_keyboard_but_mods();
+                    register_code(KC_V);
+                } else {
+                    unregister_code(KC_V);
+                }
+            } else {
+                if (record->event.pressed) {
+                    register_code(KC_D);
+                } else {
+                    unregister_code(KC_D);
+                }
+            }
+            return false;
+            break;
+
+        return false;
+        break;
     }
+    /* Any clever remapping with modifiers should happen here e.g. shift bablkey does opposite*/
+#ifdef USE_BABLPASTE
+    if( keycode >= BABL_START_NUM && id < (BABL_START_NUM + BABL_NUM_MACROS ) ) {
+        if (record->event.pressed)  { // is there a case where this isn't desired?
+            babblePaste ( record,  keycode);
+            return false;
+        }
+    }
+#endif
 
+    return true;
+}
 
-return MACRO_NONE;
-};
-
-
-
+void keyboard_post_init_user(void) {
+  // Customise these values to desired behaviour
+  // debug_enable=true;
+  //debug_matrix=true;
+  //debug_keyboard=true;
+  //debug_mouse=true;
+}
 
 
 void matrix_init_user(void) {
+    #ifdef RGB_MATRIX_ENABLE
+    #ifdef RGB_DI_PIN
+        rgblight_setrgb(RGB_GREEN);
```

Comment:
This is more of a pet peeve, and can be ignored:
```suggestion
        rgb_matrix_set_color_all(RGB_GREEN);
```

Since this is the rgb matrix function, as per the preprocessor, the rgb_matrix function should be used.

Reply:
whoops, should have been RGBLIGHT_ENABLE , not the matrix. Fixed.